### PR TITLE
Add ability to add multiple medical conditions

### LINKED ISSUE #95: Update addMedCon to take multiple medical conditions
So that doctors can take note of multiple 
medical conditions that patients might have

### PR DESCRIPTION
**Pull request by TechRaj**
Fixes #95

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddMedConCommandParser.java [hunk 1]
```diff
@@ -35,10 +38,16 @@ public AddMedConCommand parse(String args) throws ParseException {
...
         Nric nric = ParserUtil.parseNric(nricStr);
 
-        // Parse MedCon if present, otherwise use empty value
-        String medConStr = argMultimap.getValue(PREFIX_MEDCON).orElse("");
-        MedCon medCon = new MedCon(medConStr);
+        // Parse all MedCon values and add them to a set
+        Set<MedCon> medCons = new HashSet<>();
+        for (String medConStr : argMultimap.getAllValues(PREFIX_MEDCON)) {
+            if (medConStr.length() > 45) {
+                throw new ParseException("Medical condition length exceeds the limit of 45 characters");
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/parser/AddMedConCommandParser.java:45]: Just a nit: Could assign this to a variable called `MESSAGE_CONSTRAINTS_LENGTH`

File: src/test/java/seedu/address/logic/parser/AddMedConCommandParserTest.java [hunk 2]
```diff
@@ -65,5 +82,15 @@ public void parse_compulsoryFieldsMissing_failure() {
         assertParseFailure(parser, MEDCON_DESC_AMY, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                 AddMedConCommand.MESSAGE_USAGE));
     }
+
+
+    @Test
+    public void parse_medConLengthGreaterThan45Characters_failure() {
+        // Medical condition exceeds 45 characters
+        String input = NRIC_DESC_AMY + " c/" + INVALID_MEDCON;
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/logic/parser/AddMedConCommandParserTest.java:90]: Could use `PREFIX_MEDCON`

File: src/main/java/seedu/address/logic/commands/AddMedConCommand.java [hunk 3]
```diff
@@ -20,43 +22,52 @@ public class AddMedConCommand extends Command {
...
             + PREFIX_NRIC + "T0123456F "
-            + "c/Diabetes";
+            + "c/Diabetes c/Hypertension";
 
     public static final String MESSAGE_ARGUMENTS = "NRIC: %1$d, MedCon: %2$s";
 
-    public static final String MESSAGE_ADD_MEDCON_SUCCESS = "Added medical conditon to Person: %1$s";
-    public static final String MESSAGE_DELETE_MEDCON_SUCCESS = "Removed medical conditon from Person: %1$s";
+    public static final String MESSAGE_ADD_MEDCON_SUCCESS = "Added medical condition to Person: %1$s";
+    public static final String MESSAGE_DELETE_MEDCON_SUCCESS = "Removed medical condition from Person: %1$s";
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/commands/AddMedConCommand.java:36]: Remember to place this in 'DeleteMedCon` once the class is created

**Review by yong-zhuo — APPROVED** [review]
Overall, LGTM 👍🏻

File: src/main/java/seedu/address/logic/commands/AddApptCommand.java [hunk 4]
```diff
@@ -121,7 +121,7 @@ private static Person createEditedPerson(Person personToEdit,
         Gender updatedGender = personToEdit.getGender();
         Nric updatedNric = personToEdit.getNric();
         Set<Tag> updatedTags = personToEdit.getTags();
-        MedCon updatedMedCon = personToEdit.getMedCon();
+        Set<MedCon> updatedMedCon = personToEdit.getMedCons();
```
Comment (hweinian) [inline src/main/java/seedu/address/logic/commands/AddApptCommand.java:124]: Good use of Set

File: src/main/java/seedu/address/logic/commands/AddMedConCommand.java [hunk 5]
```diff
@@ -20,43 +22,52 @@ public class AddMedConCommand extends Command {
...
 
     private final Nric nric;
-    private final MedCon medCon;
+    private final Set<MedCon> medCons;
 
     /**
-     * @param nric of the patient to assign the medical condition to
-     * @param medCon of the patient
+     * @param nric of the patient to assign the medical conditions to
+     * @param medCons set of medical conditions to be assigned
```
Comment (hweinian) [inline src/main/java/seedu/address/logic/commands/AddMedConCommand.java:45]: Could include punctuation

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 6]
```diff
@@ -247,12 +247,21 @@ public Optional<Nric> getNric() {
         public void setNric(Nric nric) {
             this.nric = nric;
         }
-        public Optional<MedCon> getMedCon() {
-            return Optional.ofNullable(medCon);
+        /**
+         * Returns an unmodifiable medCon set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code medCons} is null.
+         */
```
Comment (hweinian) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:254]: Missing space. Javadocs is optional for getters

**Review by hweinian — APPROVED** [review]
LGTM. Just fix javadocs

**Pull request merged**